### DISCUSSION TITLE
migrate data  from blob storage on cephfs to s3 online?

### DISCUSSION THREAD

**Discussion by zhucan** [distribution/distribution#4198] [Ideas]
Beacuse of  our backend  storage is cephfs with replica 3, it need to much storage space to store data; so we want to change the banckend storage from cephfs to s3(EC: 8+2), to reduce storage space waste. is there way to support it?

**Comment by milosgajdos** [comment]
You can use something like https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane.md to copy the data over, but you'd have to mark the source read only before the data is copied over to the new storage backend.

**Reply by zhucan** [reply]
only support offline? @milosgajdos

**Reply by milosgajdos** [reply]
Can you define what you mean by "online"? Do you mean by that that the registry is accepting writes whilst you are migrating data into new storage? This won't be ideal, will it.

Data migrations are hard. If you want make sure consistency between the two data stores, you need to make one as read-only, copy the data over to the new one and then switch over. Otherwise you might find yourself in the situation where some data is in X and not in Y.

**Comment by zhucan** [comment]
because there is so many data on the storage,  we can only accept stop taking service briefly, so whether it's possible copy the data from  src to des not mark the src to read only, after most of part data was copied, now top the server, then run copy again to migrate the  the latest data?

**Reply by milosgajdos** [reply]
you can try doing that somehow but it's going to be painful. I've never done that myself.

**Reply by zhucan** [reply]
thanks, I will  try it.
